EndpointHandler.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+require "settings.php";
+
+// Decodes the data array and returns the value of "claim"
+function getClaim($claim, $data) {
+	
+	$data_array = json_decode($data, true);
+	return $data_array[$claim];
+}
+
+// A class to handle both fetching and sending data to the various endpoints
+class EndpointHandler {
+	
+	private $metadata = "";
+	
+	public function __construct($policy_name) {
+		$this->getMetadata($policy_name);
+	}
+	
+	// Fetches the data at an endpoint using a HTTP GET request
+	public function getEndpointData($uri) {
+
+		$ch = curl_init($uri);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); 
+		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+		$resp = curl_exec($ch);
+		curl_close($ch);
+		return $resp;
+	}
+	
+	// Using a HTTP POST request, sends data to the endpoint and receives the response
+	public function postEndpointData($uri, $fields) {
+		
+		//url-ify the data for the POST
+		$fields_string = "";
+		foreach ($fields as $key=>$value) { $fields_string .= $key.'='.$value.'&'; }
+		$fields_string = rtrim($fields_string, '&');
+					
+		$ch = curl_init();
+		curl_setopt($ch,CURLOPT_URL, $uri);
+		curl_setopt($ch,CURLOPT_POST, sizeof($fields));
+		curl_setopt($ch,CURLOPT_POSTFIELDS, $fields_string);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); 
+		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+
+		$resp = curl_exec($ch);
+		curl_close($ch);
+		return $resp;
```

Comment:
cURL (and other similar libraries) is notoriously difficult to use in a cross-platform manner in a PHP plugin. Instead, WP folks have been kind enough to give us `wp_remote_post` instead which takes care of all the cross-platform dirty work: [https://developer.wordpress.org/reference/functions/wp_remote_post/](https://developer.wordpress.org/reference/functions/wp_remote_post/)